harness/src/account_store.rs

```
@@ -0,0 +1,18 @@
+//! A trait for implementing an account store, to be used with the
+/// `MolluskWithAccountStore`.
+use {solana_account::Account, solana_pubkey::Pubkey};
+
+/// A trait for implementing an account store, to be used with the
+/// `MolluskWithAccountStore`.
+pub trait AccountStore {
+    /// Returns the default account to be used when an account is not found.
+    fn default_account(&self, _pubkey: &Pubkey) -> Account {
+        Account::default()
+    }
```

Should we have a fallback like this at all? It feels like this could swallow the error of origin and later the consumer may get an somewhat unrelated error message.

---

I think we should, only because I think we should make the base behavior operate just like the Solana runtime. If an account isn't found in AccountsDB, it automatically defaults to a system account with zero lamports, zero space, zero data, etc.

If we don't, then the harness would just panic, and you'd have to go manually insert some kind of default or mock account.

---

Oh interesting. Yes, makes sense to match the runtime behavior. Wonder why it does that versus throwing a specific AccountNotFound error.

---

Mostly so you can do things like transfer to an address that isn't "registered" with the chain yet.